### ISSUE TITLE
Build this on top of javax.websocket-api

### ISSUE THREAD

**Issue by alexm77** [ChargeTimeEU/Java-OCA-OCPP#60]
Building on top of the standard API for WebSockets in java would be safer and more flexible. It complicates the deployment a bit, because of the need to supply the implementation classes, but overall it should be a better solution.

**Comment by TVolden** [comment]
Hi @alexm77

Thanks for your request. I'll look into it. I'm thinking about exporting soe of the 3rd party dependencies into their own modules, so it's easier to mix and match.

I have allot to see to these next days but hopefully I have something to report by the end of the week.

Thomas

**Comment by alexm77** [comment]
All that's needed is [https://mvnrepository.com/artifact/javax.websocket/javax.websocket-api/1.1](url)
And then we can add tyrus and whatnot at runtime. I will look into it, but I can't promise I can scrape the time to put a pull request together.

**Comment by TVolden** [comment]
Hi again,

I'm sorry to say, I haven't had much of a chance to play around with it. I found some examples I would like to go with, but it would require changes to the exposed API. I'd like to have a 0.5 release with the current websocket lib, before I change too much. I'm considering making a 0.5 release branch.

Examples for later: https://stackoverflow.com/questions/26452903/javax-websocket-client-simple-example

**Comment by alexm77** [comment]
Yes, I've spent some time on it and I have seen there's a lot to change.
It's pretty easy to work with javax-websocket, but it's a bit different
from what the current library does.

**Comment by dersteve** [comment]
@alexm77 Are you still interested in doing some work on this issue? I would like to use this library together with Spring and Tomcat (as a Server application where chargepoints connect to) and it seems with some modularisation and swapping out the underlying web-socket implementation with javax.websocket we could reach this step together.

**Comment by alexm77** [comment]
I've looked into this and it's just too much work. You need to almost
rewrite the entire thing for it to work like the JSR does.
That said, it's not a herculean task or anything, it can probably be done
in a couple of weeks. It's just that I'm not sure I have that much time to
spare.

**Comment by marcinar** [comment]
You can work around this without changes in the library. I managed to integrate it with Spring WebSocket and Tomcat's implementation of JSR-356.

I can't share the code, but the basic idea is that you make a dummy implementation of eu.chargetime.ocpp.Listener that saves and exposes the ListenerEvents parameter of the open method. You can then pass it to eu.chargetime.ocpp.Server, and call the open method of the server (with some dummy hostname and port), which allows you to retrieve ListenerEvents from your listener.

You can use this object to signal new sessions and messages just like WebSocketListener does. Note that you will need to extend WebSocketReceiver and intercept the RadioEvents parameter of the accept method since the relay method is package-private.

It's not exactly simple, but at least you don't need to use reflection.

**Comment by frgo** [comment]
Hi @marcinar , I'm also trying to use the lib on a Spring project, the lib works fine when using a different ports for WebSocket and Tomcat. However this is bringing some headaches to deploy the project as an Azure AppService, it seems Azure AppService doesn't allow to expose an aditional port.

I tryied to follow your sugestions to extend the lib, but wasn't able to make it work. Would be possible for you to share just some reference code?

Thanks in advance

**Comment by kalosha** [comment]
Hi @frgo 
Could you share with us same example project using Spring?
Thanks in advance

**Comment by robert-s-ubi** [comment]
No activity on this for years, so I'm closing this issue.
